packages/web3-constants/evm/trader.json

```diff
@@ -1018,5 +1018,85 @@
         "Fantom": "",
         "Aurora": "0x754e1d90e536e4c1df81b7f030f47b4ca80c87120e145c294f098c83a6cb5ace",
         "Aurora_Testnet": ""
+    },
+    "MDEX_ROUTER_ADDRESS": {
+        "Mainnet": "0x74119c3bca85bEA0538A62319a79b4a372590B47",
+        "Ropsten": "",
+        "Rinkeby": "",
+        "Kovan": "",
+        "Gorli": "",
+        "BSC": "0x0384E9ad329396C3A6A401243Ca71633B2bC4333",
+        "BSCT": "",
+        "Matic": "",
+        "Mumbai": "",
+        "Arbitrum": "",
+        "Arbitrum_Rinkeby": "",
+        "xDai": "",
+        "Avalanche": "",
+        "Avalanche_Fuji": "",
+        "Celo": "",
+        "Fantom": "",
+        "Aurora": "",
+        "Aurora_Testnet": ""
+    },
+    "MDEX_FACTORY_ADDRESS": {
+        "Mainnet": "0x7DAe51BD3E3376B8c7c4900E9107f12Be3AF1bA8",
+        "Ropsten": "",
+        "Rinkeby": "",
+        "Kovan": "",
+        "Gorli": "",
+        "BSC": "0x3CD1C46068dAEa5Ebb0d3f55F6915B10648062B8",
+        "BSCT": "",
+        "Matic": "",
```

Comment:
Could you provide the source of these contract address? Including the `INIT_CODE_HASH`.  Where do you get them from? We need to verify them to make sure they are official contracts.

Reply:
Heco Network -> https://github.com/mdexSwap/contracts
Binance -> https://github.com/mdexSwap/bscswap

Reply:
I don't have Heco yet.
I have checked https://github.com/mdexSwap/bscswap for BSC too. What about for Mainnet?

Reply:
@UncleBill check on deployer: https://etherscan.io/address/0xc6f3b19075f20ad4e9efc7662ac2631d90aa7226
for binance and mainnet is the same implementer

Reply:
OK. I can verify them now.